pyquil/paulis.py

```diff
@@ -69,19 +73,19 @@ class PauliTerm(object):
     """A term is a product of Pauli operators operating on different qubits.
     """
 
-    def __init__(self, op, index, coefficient=1.0):
+    def __init__(self, op: str, index: int, coefficient: Union[int, float, complex] = 1):
```

Comment:
I'll revert this. This PR originally included lots of changes to duck-type the pauliterm coefficient so you could leave it as e.g. an integer instead of it always being promoted to `complex`. I removed all those changes from this PR (but it's still something I'd like to do eventually!)